src/nodes.coffee

```diff
@@ -1639,7 +1639,8 @@ exports.Op = class Op extends Base
     not @second
 
   isComplex: ->
-    not (@isUnary() and @operator in ['+', '-']) or @first.isComplex()
+    not (@isUnary() and @operator in ['+', '-'] and
+         @first instanceof Value and @first.isSimpleNumber())
```

Comment:
If you mean that I should change the line to `not (@isUnary() and @operator in ['+', '-'] and not @first.isComplex()`, then that is a problem because for example `a` is considered non-complex and the whole point of this PR is that for example `+a` should _be_ considered complex, so that a `_ref` is generated for it.